spark/src/main/java/org/apache/iceberg/spark/data/SparkParquetWriters.java

```diff
@@ -115,7 +115,7 @@ private SparkParquetWriters() {
       return new MapDataWriter<>(repeatedD, repeatedR,
           newOption(repeatedKeyValue.getType(0), keyWriter),
           newOption(repeatedKeyValue.getType(1), valueWriter),
-          sMap.keyType(), sMap.valueType());
+          ((MapType) sMap).keyType(), ((MapType) sMap).valueType());
```

Comment:
Can we use `arrayElementType` `mapKeyType` `mapValueType` etc... to avoid casting?

Reply:
Done.